File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/api/idea/util/MarkdownHelper.java
package org.ifinalframework.jetbrains.plugins.aio.api.idea.util;


import com.google.inject.ImplementedBy;
import com.intellij.psi.PsiElement;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * MarkdownHelper
 *
 * @author iimik
 * @since 0.0.1
 **/
@ImplementedBy(DefaultMarkdownHelper.class)
public interface MarkdownHelper {

    @Nullable
    String getMarkdownPath(@NotNull PsiElement element);

}

File: idea-plugin/src/main/kotlin/org/ifinalframework/jetbrains/plugins/aio/api/markdown/MarkdownHelper.kt
package org.ifinalframework.jetbrains.plugins.aio.api.markdown

import com.google.inject.ImplementedBy
import com.intellij.psi.PsiElement


/**
 * MarkdownHelper
 *
 * @author iimik
 * @since 0.0.1
 **/
@ImplementedBy(DefaultMarkdownHelper::class)
interface MarkdownHelper {
    fun getMarkdownPath(element: PsiElement): String?
}